packages/vscode-extension-dashbuilder-editor/src/extension/ComponentsHttpServer.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { LocalHttpServer } from "@kie-tools-core/backend/dist/api";
+import { getPortPromise } from "portfinder";
+import * as http from "http";
+import * as fs from "fs";
+import * as path from "path";
+
+export class ComponentServer extends LocalHttpServer {
+  private componentsPath: string;
+  private server: http.Server;
+
+  requestListener = (request: any, response: any) => {
+    if (request.url == "/" || request.url == "" || request.url == "index.html") {
+      response.writeHead(200);
+      response.end("Components server is alive! Base path is " + this.componentsPath);
+      return;
+    }
+
+    var userInput = path.normalize(request.url).replace(/^(\.\.(\/|\\|$))+/, "");
+    var filePath = path.join(this.componentsPath, userInput);
+    if (filePath.indexOf(this.componentsPath) !== 0) {
+      console.debug("Denying access to file " + filePath);
+      response.writeHead(403);
+      return;
+    }
+    console.debug("Requesting file: " + filePath);
+
+    fs.readFile(filePath, function (error, content) {
+      if (error) {
+        if (error.code == "ENOENT") {
+          response.writeHead(404);
+          response.end();
+        } else {
+          response.writeHead(500);
+          response.end();
+        }
+      } else {
+        response.writeHead(200);
+        response.end(content, "utf-8");
+      }
+    });
+  };
+
+  constructor(componentsPath: string) {
+    super();
+    this.componentsPath = componentsPath;
+  }
```

Comment:
```suggestion
  constructor(private readonly componentsPath: string) {
    super();
  }
```
Optional to make it shorter. You can also remove line 24.

Reply:
TS HACKER²